enhancements/upgrade-bundle.md

```diff
@@ -0,0 +1,260 @@
+
+  * Operator fails to come up after creating subscription 
+    * If operator fails to come up after creating the subscription, it failed to create a second install plan and the SubscriptionStatus will indicate that. Therefore, we proceed to logging the error message and exiting. 
+
+  * What happens when a catalog source is not created?
+    * When a catalog source fails to create, it would show on the status of the CatalogSource. We log an error message, and exit. 
+
+  * Do we cleanup resources after failures or not? 
+    * We could offer flexibility to the operator developers to use a flag --skip-cleanup to indicate whether or not cleanup of resources is desired. This could enable the operator developers to capture all the logs and inspect the pod failures.
+
+  * Behavior of `run bundle-upgrade` in the presence of an OperatorGroup
+    * It is important to note that the presence of an OperatorGroup in the namespace can affect the how `run bundle-upgrade` operates. The presence of the OperatorGroup will take precedence. If the OperatorGroup does not support  version an error condition will occur. The `run bundle-upgrade` command will return with a non-zero exit status and log the errors. 
+
+  * Install Mode and OperatorGroups
+    * We support 3 modes:
+      * AllNamespaces
+      * OwnNamespace
+      * SingleNamespace=ns1
+        * NOTE: if ns1 is the same as the operator's namespace, a warning that "ns1 can not be the same as the operator's namespace" will be printed
+    * When `--install-mode` is not supplied, the selection algorithm is as follows:
+      * If an `OperatorGroup` already exists with the correct namespacing, use it.
+      * Else, create an `OperatorGroup` with the following preferences:
+    	  * If `AllNamespaces` is supported, use this mode.
+    	  * Else if `OwnNamespace` is supported, use this mode.
+    	  * Else:
+			  * If `SingleNamespace` is not supported, return an error saying "no valid install mode is supported"
+			  * Else return an error saying "--install-mode=SingleNamespace=<namespace> is required"
+
+Example UX Scenarios are discussed below. 
+
+#### Command Line Interface
+The new command line will look as follows:
+
+```
+operator-sdk run bundle-upgrade (alias: bu) <bundle-image> --index-image=<> --starting-csv=<> [--channel=<>] [--operator-namespace=<>] [--install-mode=<>]
+```
+
+* Positional arguments:
+    * `bundle-image`
+* Required flags:
+    * `--index-image` 
+    * `--starting-csv`
+* Optional flags:
+    * `--channel` 
+    * `--operator-namespace`
+    * `--install-mode`
+
+* `<bundle-image>` is a positional argument that specifies the bundle image that we want to upgrade to
+* `--index-image=<>` is a required flag, which contains the operator we’re upgrading from
+* `--starting-csv=<>` is a required flag, which specifies the first version of the CSV that OLM will install as part of the subscription for a given package. 
```

Comment:
nit: since these are required, should the be positional parameters instead of arguments?